### PR TITLE
Fixes Time Series Cancellation Test

### LINKED ISSUE #84095: [CI] SearchCancellationIT testCancellationDuringTimeSeriesAggregation failing
Perhaps in this run the search was cancelled too early? Doesn't reproduce for me anyway so seems like some kind of race.

**Build scan:**
https://gradle-enterprise.elastic.co/s/gxwsmk4hxf5y2/tests/:server:internalClusterTest/org.elasticsearch.search.SearchCancellationIT/testCancellationDuringTimeSeriesAggregation

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.SearchCancellationIT.testCancellationDuringTimeSeriesAggregation" -Dtests.seed=B858CAC59575905 -Dtests.locale=sr-RS -Dtests.timezone=America/Catamarca -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.SearchCancellationIT&tests.test=testCancellationDuringTimeSeriesAggregation

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a string containing "TimeSeriesIndexSearcher"
     but: was "Failed to execute phase [fetch], ; shardFailures {[5R6jMFYWQhyExkaRiv8xpA][test][0]: org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:36497][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.search.query.QueryPhaseExecutionException: Query Failed [Failed to execute main query]
	at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:205)
	at org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:88)
	at org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:462)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:625)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:487)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.elasticsearch.tasks.TaskCancelledException: task cancelled [by user request]
	at org.elasticsearch.tasks.CancellableTask.getTaskCancelledException(CancellableTask.java:121)
	at org.elasticsearch.tasks.CancellableTask.ensureNotCancelled(CancellableTask.java:97)
	at org.elasticsearch.search.DefaultSearchContext.lambda$preProcess$0(DefaultSearchContext.java:269)
	at org.elasticsearch.search.internal.ContextIndexSearcher$MutableQueryTimeout.checkCancelled(ContextIndexSearcher.java:337)
	at org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:178)
	at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:167)
	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:541)
	at org.elasticsearch.search.query.QueryPhase.searchWithCollector(QueryPhase.java:232)
	at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:187)
	... 13 more
}{[5R6jMFYWQhyExkaRiv8xpA][test][2]: org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:36497][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.tasks.TaskCancelledException: task cancelled [by user request]
	at org.elasticsearch.tasks.CancellableTask.getTaskCancelledException(CancellableTask.java:121)
	at org.elasticsearch.tasks.CancellableTask.ensureNotCancelled(CancellableTask.java:97)
	at org.elasticsearch.search.SearchService.checkCancelled(SearchService.java:814)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:965)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:621)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:487)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
}{[5R6jMFYWQhyExkaRiv8xpA][test][3]: org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:36497][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.tasks.TaskCancelledException: task cancelled [by user request]
	at org.elasticsearch.tasks.CancellableTask.getTaskCancelledException(CancellableTask.java:121)
	at org.elasticsearch.tasks.CancellableTask.ensureNotCancelled(CancellableTask.java:97)
	at org.elasticsearch.search.SearchService.checkCancelled(SearchService.java:814)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:965)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:621)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:487)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
}
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:716)
	at org.elasticsearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:103)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.elasticsearch.tasks.TaskCancelledException: Cancelled
	at org.elasticsearch.search.aggregations.AggregationReduceContext.consumeBucketsAndMaybeBreak(AggregationReduceContext.java:117)
	at org.elasticsearch.search.aggregations.timeseries.InternalTimeSeries.reduce(InternalTimeSeries.java:217)
	at org.elasticsearch.search.aggregations.InternalAggregations.reduce(InternalAggregations.java:155)
	at org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:102)
	at org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:532)
	at org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:501)
	at org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:132)
	at org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:112)
	at org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:98)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	... 6 more
"

  at __randomizedtesting.SeedInfo.seed([B858CAC59575905:89ED0086CDB94051]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.search.SearchCancellationIT.testCancellationDuringTimeSeriesAggregation(SearchCancellationIT.java:330)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by imotov**
Ensures that we have enough documents in each shard to trigger the cancellation
check. Fixes the issue with uneven document distribution due to bad routing and
increases number of documents to be higher than CHECK_CANCELLED_SCORER_INTERVAL.

Fixes #84095

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java [hunk 1]
```diff
@@ -291,7 +293,7 @@ public void testCancellationDuringTimeSeriesAggregation() throws Exception {
                 bulkRequestBuilder.add(
                     client().prepareIndex("test")
                         .setOpType(DocWriteRequest.OpType.CREATE)
-                        .setSource("@timestamp", now + (long) i * numberOfDocsPerRefresh + j, "val", (double) j, "dim", String.valueOf(i))
+                        .setSource("@timestamp", now + (long) i * numberOfDocsPerRefresh + j, "val", (double) j, "dim", String.valueOf(j))
```
Comment (nik9000) [inline server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java:296]: So we end up with thousands of dimensions? Do we want that shape here?

File: server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java [hunk 2]
```diff
@@ -263,7 +263,9 @@ public void testCancellationDuringTimeSeriesAggregation() throws Exception {
         int numberOfShards = between(2, 5);
         long now = Instant.now().toEpochMilli();
         int numberOfRefreshes = between(1, 5);
-        int numberOfDocsPerRefresh = numberOfShards * between(1500, 2000) / numberOfRefreshes;
+        // After a few initial checks we check every 2048 - number of shards records so we need to ensure all
+        // shards have enough records to trigger a check
+        int numberOfDocsPerRefresh = numberOfShards * between(3000, 3500) / numberOfRefreshes;
```
Comment (nik9000) [inline server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java:268]: :+1:

File: server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java (hunk 1 above)
Comment (imotov) [reply server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java:296]: I don't think it hurts, but I will reduce it to 100 dimensions.

Comment (nik9000) [reply server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java]: You could randomize it if you want. I'm ok with whatever. So long as I know the actual fix is the other part.

**Review by nik9000 — APPROVED** [review]

**Comment by imotov** [comment]
@elasticmachine run elasticsearch-ci/part-1

File: server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java (hunk 1 above)
Comment (imotov) [reply server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java]: The number of shards and document is already randomized, so it should be ok. The goal here is to ensure we have enough different ids to ensure more or less equal distribution of documents across the shards.

**Pull request merged**